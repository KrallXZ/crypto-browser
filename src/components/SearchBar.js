import React from 'react'
import Input from 'material-ui/Input'

export default class SearchBar extends React.PureComponent {
  render() {
    return (
      <Input
        placeholder="Search"
        value={this.props.value}
        onChange={this.props.callback}
      />
    )
  }
}
