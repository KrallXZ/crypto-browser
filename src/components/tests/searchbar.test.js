import React from 'react'
import SearchBar from '../SearchBar'
import renderer from 'react-test-renderer'

describe('SearchBar', () => {
  it('renders correctly', () => {
    const component = renderer.create(<SearchBar />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders as input', () => {
    const component = renderer.create(<SearchBar />)
    let tree = component.toJSON()
    expect(tree.children[0].type).toBe('input')
  })
})


