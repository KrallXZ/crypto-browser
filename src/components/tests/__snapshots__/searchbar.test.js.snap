// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar renders correctly 1`] = `
<div
  className="MuiInput-root-1 MuiInput-inkbar-3 MuiInput-underline-7"
  onBlur={[Function]}
  onFocus={[Function]}
>
  <input
    aria-invalid={undefined}
    aria-required={undefined}
    autoComplete={undefined}
    autoFocus={undefined}
    className="MuiInput-input-10"
    defaultValue={undefined}
    disabled={undefined}
    id={undefined}
    name={undefined}
    onChange={[Function]}
    onKeyDown={undefined}
    onKeyUp={undefined}
    placeholder="Search"
    readOnly={undefined}
    required={undefined}
    rows={undefined}
    type="text"
    value={undefined}
  />
</div>
`;
